File: Data Structures & Algorithms/non-cyclical-number/submission-0.java
class Solution {
    public boolean isHappy(int n) {
        Set<Integer> visited = new HashSet<>();
        while (n != 1) {
            if (visited.contains(n)) return false;
            visited.add(n);
            String str = String.valueOf(n);
            int sum = 0;
            for (char c : str.toCharArray()) {
                sum += (c - '0') * (c - '0');
            }
            n = sum;
        }
        return true;
    }
}

File: Data Structures & Algorithms/score-of-a-string/submission-0.cpp
class Solution {
public:
    int scoreOfString(string s) {
        int sum = 0;
        for (int i = 0; i < s.size() - 1; i++) {
            sum += abs(static_cast<int>(s[i + 1]) - static_cast<int>(s[i]));
        }
        return sum;
    }
};